### PR TITLE
Add new API to get the config last modification time

### PR DESCRIPTION
**Pull request by jenting**
#### Propose Changes

- Rename backup volume config and backup config configuration-related code to use the word `config` instead of `metadata` to make it more clear.
- `backup config-metadata`: a new command to get the config (backup volume config _or_ backup config) metadata to get the last modification time. This could prevent the longhorn manager read the config periodically even the config did not change.

#### Issue

https://github.com/longhorn/longhorn/issues/1761

### CODE REVIEW

**Review by shuo-wu — COMMENTED** [review]
The naming is weird... The config file store the meta info of a backup or a backup volume. But here `metadata` means the meta info of the config file itself and it reports the last modified time only...

**Comment by jenting** [comment]
> The naming is weird... The config file store the meta info of a backup or a backup volume. But here `metadata` means the meta info of the config file itself and it reports the last modified time only...

If using the `head` naming, would it more intuitive? https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD

File: cmd/head.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"github.com/urfave/cli"
+
+	"github.com/longhorn/backupstore"
+	"github.com/longhorn/backupstore/util"
+)
+
+func GetConfigMetadataCmd() cli.Command {
+	return cli.Command{
+		Name:   "head",
+		Usage:  "get the config metadata",
```
Comment (shuo-wu) [inline cmd/head.go:15]: `... this returns the last modification time of a config file for now.`

File: backupstore.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func removeVolume(volumeName string, driver BackupStoreDriver) error {
 	return nil
 }
 
-func EncodeMetadataURL(backupName, volumeName, destURL string) string {
+func EncodeConfigURL(backupName, volumeName, destURL string) string {
```
Comment (shuo-wu) [inline backupstore.go:99]: `EncodeBackupURL`. Similar below

**Review by shuo-wu — CHANGES_REQUESTED** [review]

File: cmd/head.go (hunk 1 above)
Comment (jenting) [reply cmd/head.go]: Add to the `description`.

**Review by shuo-wu — APPROVED** [review]
LGTM. Remember to rearrange the commits

**Pull request merged**